src/lib/Builder/NotificationBuilder.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Builder;
+
+use Twig\Environment;
+
+/**
+ * Builder for notification messages, returns output from twig.
+ */
+class NotificationBuilder
+{
+    const TEMPLATE = 'EzSystemsHybridPlatformUiBundle::notification.html.twig';
+
+    const DEFAULT_TIMEOUT = 10;
+
+    const TYPE_SUCCESS = 'positive';
+
+    const TYPE_ERROR = 'error';
+
+    /**
+     * @var Environment
+     */
+    private $environment;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var string
+     */
+    private $message;
+
+    /**
+     * @var int
+     */
+    private $timeout = self::DEFAULT_TIMEOUT;
+
+    /**
+     * @var bool
+     */
+    private $copyable;
+
+    /**
+     * @var string
+     */
+    private $details;
+
+    public function __construct(Environment $environment)
+    {
+        $this->environment = $environment;
+    }
+
+    /**
+     * Sets notification type as success.
+     *
+     * @return $this
+     */
+    public function setSuccess()
+    {
+        $this->type = self::TYPE_SUCCESS;
+
+        return $this;
+    }
+
+    /**
+     * Sets notification type as error.
+     *
+     * @return $this
+     */
+    public function setError()
+    {
+        $this->type = self::TYPE_ERROR;
+
+        return $this;
+    }
+
+    /**
+     * Sets error details.
+     *
+     * @param string $details
+     *
+     * @return $this
+     */
+    public function setErrorDetails(string $details)
+    {
+        $this->copyable = true;
+        $this->details = $details;
+
+        return $this;
+    }
+
+    /**
+     * Sets notification message.
+     *
+     * @param string $message
+     *
+     * @return $this
+     */
+    public function setMessage(string $message)
+    {
+        $this->message = $message;
+
+        return $this;
+    }
+
+    /**
+     * Sets notification timeout.
+     *
+     * @param int $timeout
+     *
+     * @return $this
+     */
+    public function setTimeout(int $timeout)
+    {
+        $this->timeout = $timeout;
+
+        return $this;
+    }
+
+    /**
+     * Builds notification message.
+     *
+     * @return string
+     */
+    public function getResult()
```

Comment:
you are coupling the Notification creation and rendering and I think it's a mistake for 2 reasons:

1. Notifications won't always be rendered the same way. We already render some notifications in Hybrid Platform UI (when coming from PJAX response handled by PlatformUIBundle) and those are set on the app [to be serialized in JSON](https://github.com/ezsystems/hybrid-platform-ui/blob/master/src/lib/Components/App.php#L85). @bdunogier is also setting up pretty much the same [in the PR about error handling](https://github.com/ezsystems/hybrid-platform-ui/pull/68/files#diff-d4bd0f0dbb343592a25da1345613ae69R70).
2. The creation of a Notification and its rendering is usually done very separately. For instance, after a form submit, an action is done which ends up with one or several notifications, then you are redirected and after that notifications are rendered.

So for me, we should first have a `Notification` value object (which could be extended to have for instance an `ErrorNotification` where `timeout` is forced to 0) to represent the concept of Notification. Then how those are rendered depends on where there are used.

For tabs, I think each tab should store its own notifications in its own entry in the flash bag. So the default operation flow in tabs should be the following:

1. the user submits a form in a tab
2. the form is handled, one or several notifications are put in a flash bag entry dedicated to the tab, then a redirection to the tab action is generated
3. when rendering the tab, the tab flash bag entry is read to render the `ez-notification` elements along the tab content

Now remains the case of Show/Hide Location (and possibly for set main operation) where there should be no redirection to the tab view. For those, I guess after processing the form, the redirection could be made to a very simple action that would respond with an HTTP status corresponding to what happened while processing the form and would just read the corresponding flash bag.